src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyExecutionManager.cs

```diff
@@ -96,7 +98,10 @@ public virtual int StartTestRun(TestRunCriteria testRunCriteria, ITestRunEventsH
                     testSources = testRunCriteria.Tests.GroupBy(tc => tc.Source).Select(g => g.Key);
                 }
 
-                this.isCommunicationEstablished = this.SetupChannel(testSources, this.cancellationTokenSource.Token);
+                lock (locObject)
```

Comment:
We are not using Lock now, instead we are checking cancelation token after initialization to cancel run.